src/ng/directive/form.js

```diff
@@ -131,6 +128,28 @@ function FormController(element, attrs, $scope, $animate, $interpolate) {
     if (control.$name) {
       form[control.$name] = control;
     }
+
+    control.$$parentForm = form;
+
+    forEach(control.$pending, function(value, name) {
+      form.$setValidity(name, undefined, control);
+    });
+
+    forEach(control.$error, function(value, name) {
+      form.$setValidity(name, false, control);
+    });
+
+    forEach(control.$$success, function(value, name) {
+      form.$setValidity(name, true, control);
+    });
+
+    if (control.$dirty && form.$pristine) {
+      form.$setDirty();
+    }
+
+    if (control.$submitted && !form.$submitted) {
+      form.$setSubmitted();
```

Comment:
I guess this is in accordance to what happens in `$setSubmitted()` (where the parent form is also marked as submitted), but it feels unintuitive.

Anyway, since this is how things are, let's leave it as is.